### PR TITLE
Fix image load infowindow

### PR DESCRIPTION
**Pull request by javierarce**
This PR fixes a problem loading custom infowindow templates with images. It also refactor some of the infowindow.js code, makes the loading of the image more pleasant and reviews the infowindow specs.

PS: I've also added several examples in the example folder.

russia

### CODE REVIEW

**Comment by javierarce** [comment]
@xavijam: can you take a look, please?

**Comment by xavijam** [comment]
Could we add an example in the v4 branch for testing porpuses? please :)

**Comment by javierarce** [comment]
Sure! :)

**Comment by javierarce** [comment]
I've added a basic first example: `examples/v4-infowindow.html`.

**Comment by xavijam** [comment]
Does the hook appear with white background when image is being loaded?

**Comment by javierarce** [comment]
Oh, yes, that's one of the things I fixed too.

File: src/geo/ui/infowindow.js [hunk 1]
```diff
@@ -327,108 +329,126 @@ var Infowindow = View.extend({
...
+  },
+
   /**
    *  Get cover URL
    */
   _getCoverURL: function () {
     var content = this.model.get('content');
 
-    if (content && content.fields && content.fields.length > 0) {
+    if (this.$('.js-cover img').attr('src')) {
```
Comment (xavijam) [inline src/geo/ui/infowindow.js]: _Totally optional: I'd create a var storing the src of the img, being used twice._

**Comment by iriberri** [comment]
Awesome. I :revolving_hearts: you.

**Comment by xavijam** [comment]
This is the new infowindow @iriberri :)

**Comment by javierarce** [comment]
I've made several fixes to allow using custom loaders in the content of an infowindow. Sadly, I don't think we can use the same solution for the covers (because images could take longer to load than the fields and the infowindow template would be refreshed, without the special 'loading' flag that we add in the content)

**Comment by javierarce** [comment]
@xavijam: :point_up:  that said, all yours :)

**Comment by xavijam** [comment]
Checking it! :)

File: src/geo/ui/infowindow.js [hunk 2]
```diff
@@ -164,36 +160,28 @@ var Infowindow = View.extend({
     }
   },
 
-  _renderLoader: function () {
+  _renderCoverLoader: function () {
+    var $loader = $('<div class="CDB-Loader js-loader"></div>');
```
Comment (xavijam) [inline src/geo/ui/infowindow.js]: _Totally optional: You could create the element as $('<div>').addClass('CDB-Loader js-loader')_

Comment (javierarce) [inline src/geo/ui/infowindow.js]: :cool:

**Comment by xavijam** [comment]
I feel code looks great and it covers perfectly our new features, so :100:.
And also new examples are pure love :D.

**Comment by javierarce** [comment]
Thanks for the review, the advices & the kind words! :)

**Pull request merged**